code/Controllers/CMSMain.php

```diff
@@ -1668,9 +1678,11 @@ protected function getCreatableSubClasses(DataObject $record): array
         // Build the list of candidate children
         $cache = $this->getCreatableChildrenCache();
         $cacheKey = $this->generateChildrenCacheKey();
-        $children = $cache->get($cacheKey, []);
+        $children = [];
```

Comment:
Semi-related fix required for unit-tests, getCreatableChildrenCache() can be null (`private ?CacheInterface $creatableChildrenCache = null;`), so we need to account for that in this method.